packages/e2e.sandbox.subscribeapp/src/MapComponent.tsx

```diff
@@ -0,0 +1,177 @@
+import L from "leaflet";
+import React from "react";
+import { MapContainer, Marker, Popup, TileLayer } from "react-leaflet";
+import "leaflet/dist/leaflet.css";
+import { isOk } from "@osdk/api";
+import { __EXPERIMENTAL__NOT_SUPPORTED_YET_subscribe } from "@osdk/api/unstable";
+import iconUrl from "leaflet/dist/images/marker-icon.png";
+import iconShadowUrl from "leaflet/dist/images/marker-shadow.png";
+import { foundryClient } from "./client.js";
+import EntityPopup from "./EntityPopup.js";
+import { $Objects } from "./generatedNoCheck2/index.js";
+
+const defaultIcon = L.icon({
+  iconUrl,
+  shadowUrl: iconShadowUrl,
+});
+
+L.Marker.prototype.options.icon = defaultIcon;
+
+export interface Coordinates {
+  lat: number;
+  lng: number;
+}
+
+export interface EntityInformation {
+  id: string;
+  routeId: string;
+  nextStopId: string;
+}
+
+interface EntityInformationMap {
+  [key: string]: EntityInformation;
+}
+interface ObjectCoordinatesMap {
+  [key: string]: Coordinates;
+}
+
+const MapComponent: React.FC = () => {
+  const center: Coordinates = { lat: 40.7128, lng: -74.0060 };
+
+  const [entityInfo, setEntityInfo] = React.useState<EntityInformationMap>({});
+
+  const [objectCoordinates, setObjectCoordinates] = React.useState<
+    ObjectCoordinatesMap
+  >({});
```

Comment:
I separated these out, mainly so I could avoid potentially undefined properties. Users are definitely going to have to deal with this, and the best hope is that they're going to handle this in an optimal way.